types/vfile/index.d.ts

```diff
@@ -119,15 +117,15 @@ declare namespace vfile {
          * Place to store custom information.
          * It's OK to store custom data directly on the `vfile`, moving it to `data` gives a little more privacy.
          */
-        data: C['data'];
+        data: any;
```

Comment:
Could `unknown` safely be leveraged here?
https://blogs.msdn.microsoft.com/typescript/2018/07/12/announcing-typescript-3-0-rc/#the-unknown-type

Reply:
It depends on how we want our users to use `data`.

I'm not sure that we need to enforce users to make their own type guards. I think it can be optional for now. So users can easily adopt vfile with ts.